<a href="https://colab.research.google.com/github/jajeji32/Taller1/blob/main/taller_2_vania_ponce.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Pregunta 1.1 — Descripción de Variables
Examinar la base ceosal1 y elaborar una breve descripción de sus variables, distinguiendo entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico se emplea habitualmente esta base de datos?

In [ ]:
install.packages("wooldridge")
library(wooldridge)

set.seed(1)

data(ceosal1)
ceosal1 <- na.omit(ceosal1)

head(ceosal1)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



,salary,pcsalary,sales,roe,pcroe,ros,indus,finance,consprod,utility,lsalary,lsales
,<int>,<int>,<dbl>,<dbl>,<dbl>,<int>,<int>,<int>,<int>,<int>,<dbl>,<dbl>
1,1095,20,27595.0,14.1,106.4,191,1,0,0,0,6.998509,10.225389
2,1001,32,9958.0,10.9,-30.6,13,1,0,0,0,6.908755,9.206132
3,1122,9,6125.9,23.5,-16.3,14,1,0,0,0,7.022868,8.720281
4,578,-9,16246.0,5.9,-25.7,-21,1,0,0,0,6.359574,9.695602
5,1368,7,21783.2,13.8,-3.0,56,1,0,0,0,7.221105,9.988894
6,1145,5,6021.4,20.0,1.0,55,1,0,0,0,7.043160,8.703075


In [ ]:
str(ceosal1)
summary(ceosal1)

'data.frame':	209 obs. of  12 variables:
 $ salary  : int  1095 1001 1122 578 1368 1145 1078 1094 1237 833 ...
 $ pcsalary: int  20 32 9 -9 7 5 10 7 16 5 ...
 $ sales   : num  27595 9958 6126 16246 21783 ...
 $ roe     : num  14.1 10.9 23.5 5.9 13.8 ...
 $ pcroe   : num  106.4 -30.6 -16.3 -25.7 -3 ...
 $ ros     : int  191 13 14 -21 56 55 62 44 37 37 ...
 $ indus   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ finance : int  0 0 0 0 0 0 0 0 0 0 ...
 $ consprod: int  0 0 0 0 0 0 0 0 0 0 ...
 $ utility : int  0 0 0 0 0 0 0 0 0 0 ...
 $ lsalary : num  7 6.91 7.02 6.36 7.22 ...
 $ lsales  : num  10.23 9.21 8.72 9.7 9.99 ...
 - attr(*, "time.stamp")= chr "25 Jun 2011 23:03"


     salary         pcsalary          sales              roe       
 Min.   :  223   Min.   :-61.00   Min.   :  175.2   Min.   : 0.50  
 1st Qu.:  736   1st Qu.: -1.00   1st Qu.: 2210.3   1st Qu.:12.40  
 Median : 1039   Median :  9.00   Median : 3705.2   Median :15.50  
 Mean   : 1281   Mean   : 13.28   Mean   : 6923.8   Mean   :17.18  
 3rd Qu.: 1407   3rd Qu.: 20.00   3rd Qu.: 7177.0   3rd Qu.:20.00  
 Max.   :14822   Max.   :212.00   Max.   :97649.9   Max.   :56.30  
     pcroe            ros            indus           finance      
 Min.   :-98.9   Min.   :-58.0   Min.   :0.0000   Min.   :0.0000  
 1st Qu.:-21.2   1st Qu.: 21.0   1st Qu.:0.0000   1st Qu.:0.0000  
 Median : -3.0   Median : 52.0   Median :0.0000   Median :0.0000  
 Mean   : 10.8   Mean   : 61.8   Mean   :0.3206   Mean   :0.2201  
 3rd Qu.: 19.5   3rd Qu.: 81.0   3rd Qu.:1.0000   3rd Qu.:0.0000  
 Max.   :977.0   Max.   :418.0   Max.   :1.0000   Max.   :1.0000  
    consprod         utility          lsalary          

La base de datos ceosal1 contiene información sobre salarios de directores ejecutivos (CEO) y variables financieras de las empresas en las que trabajan. Esta base se utiliza comúnmente para analizar los determinantes del salario de los ejecutivos.

La variable dependiente es:

salary: salario del CEO.

Las variables explicativas principales son:

sales: ventas de la empresa.
roe: retorno sobre patrimonio.
ros: retorno sobre ventas.

En cuanto a la clasificación:

Variables continuas:
salary, sales, roe y ros.
Variables dummy:
indus, finance, consprod y utility, que indican el tipo de empresa.

Pregunta 1.2- Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable
sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que
sea fácilmente identificable.


In [ ]:
ceosal1$sales_mayor_promedio <- ifelse(ceosal1$sales > mean(ceosal1$sales), 1, 0)


1.2. Variable dummy

Se creó una variable dummy llamada sales_mayor_promedio, que toma valor 1 cuando las ventas de la empresa son mayores al promedio de la muestra y 0 en caso contrario. Esta variable permite distinguir entre empresas con alto nivel de ventas y aquellas con ventas menores o iguales al promedio.

Pregunta 1.3- Cree una variable de la multuplicación entre la variable dummy antes creada y la
variable indus. ¿Qué indica esta interacción?

In [ ]:
ceosal1$sales_mayor_promedio_indus <- ceosal1$sales_mayor_promedio * ceosal1$indus

1.3. Interacción con indus

Se construyó la variable sales_mayor_promedio_indus, que corresponde a la interacción entre la dummy anterior y la variable indus. Esta variable toma valor 1 únicamente cuando la empresa pertenece al sector industrial y tiene ventas superiores al promedio. Por lo tanto, permite identificar empresas industriales de gran tamaño.

Pregunta 1.4- Cree una variable de la multiplicación entre la variable continua roe y la variable
dummy antes creada. ¿Qué indica esta interacción?

In [ ]:
ceosal1$sales_mayor_promedio_roe <- ceosal1$sales_mayor_promedio * ceosal1$roe

1.4. Interacción con roe

Se creó la variable sales_mayor_promedio_roe, que corresponde a la interacción entre la dummy y la variable continua roe. Esta variable permite analizar si el efecto del retorno sobre patrimonio sobre el salario del CEO cambia en empresas con ventas superiores al promedio.

Pregunta 1.5- Cree una variable que indique si el compañia corresponde a industria (indus) ó si
finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable.

In [ ]:
ceosal1$indus_o_finance <- ceosal1$indus + ceosal1$finance

1.5. Variable indus o finance

Se generó la variable indus_o_finance, que toma valor 1 cuando la empresa pertenece al sector industrial o financiero, y 0 en caso contrario. Esta variable permite agrupar ambos sectores en una sola categoría.

Pregunta 2.1- Genere un modelo de regresión lineal, sobre alguna de las variables dependientes,
con respecto a las variables sales, roe y ros. Interprete sus resultados y la significancia de los
regresores. ¿El modelo tiene significancia global al 5%?

In [ ]:
modelo1 <- lm(salary ~ sales + roe + ros, data = ceosal1)

summary(modelo1)


Call:
lm(formula = salary ~ sales + roe + ros, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1495.0  -486.5  -242.8    98.9 13516.4 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 864.117516 228.399737   3.783 0.000203 ***
sales         0.015482   0.008954   1.729 0.085291 .  
roe          22.003312  11.516554   1.911 0.057454 .  
ros          -1.105191   1.450240  -0.762 0.446892    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1360 on 205 degrees of freedom
Multiple R-squared:  0.03191,	Adjusted R-squared:  0.01775 
F-statistic: 2.253 on 3 and 205 DF,  p-value: 0.08337


2.1. Se estimó un modelo de regresión lineal donde el salario del CEO (salary) depende de las variables sales, roe y ros.

El modelo es:

salary = β0 + β1·sales + β2·roe + β3·ros + error

Cada coeficiente mide el efecto de la variable correspondiente sobre el salario, manteniendo constantes las demás.

La significancia individual de los regresores se evalúa mediante sus p-values:

Si p < 0,05 → variable significativa
Si p > 0,05 → variable no significativa

La significancia global del modelo se evalúa mediante el estadístico F:

Si el p-value del F es menor a 0,05, el modelo es globalmente significativo.

Pregunta 2.2- Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2
y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?

In [ ]:
modelo2 <- lm(salary ~ sales + roe + ros + sales_mayor_promedio + sales_mayor_promedio_roe, data = ceosal1)

summary(modelo2)


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_mayor_promedio_roe, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
 -976.0  -469.1  -263.4    97.2 13566.3 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)   
(Intercept)              783.298711 258.064395   3.035  0.00272 **
sales                      0.006657   0.011197   0.594  0.55285   
roe                       24.472338  13.088222   1.870  0.06295 . 
ros                       -0.986904   1.457818  -0.677  0.49919   
sales_mayor_promedio     513.083213 510.134319   1.006  0.31572   
sales_mayor_promedio_roe  -9.498301  25.587585  -0.371  0.71087   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1361 on 203 degrees of freedom
Multiple R-squared:  0.04059,	Adjusted R-squared:  0.01696 
F-statistic: 1.718 on 5 and 203 DF,  p-value: 0.132


2.2. Se estimó un segundo modelo que incorpora las variables sales_mayor_promedio y sales_mayor_promedio_roe.

Este modelo permite analizar:

Si las empresas con ventas superiores al promedio presentan diferencias en el salario del CEO.
Si el efecto del retorno sobre patrimonio (roe) cambia dependiendo del nivel de ventas.

Un coeficiente positivo en sales_mayor_promedio indica que los CEO de empresas grandes ganan más.
La interacción permite capturar efectos diferenciados en función del tamaño de la empresa.



Pregunta 2.3- Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son
significativas para el modelo simultaneamente. Use significancia del 5%.

In [ ]:
anova(modelo1, modelo2)

,Res.Df,RSS,Df,Sum of Sq,F,Pr(>F)
,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
1,205,379231123,NA,NA,NA,NA
2,203,375832853,2,3398270,0.9177601,0.4010634


2.3. Se realiza un test de hipótesis para evaluar si las variables sales_mayor_promedio y sales_mayor_promedio_roe son significativas simultáneamente.

Hipótesis:

H0: β4 = β5 = 0
H1: al menos uno ≠ 0

Regla de decisión:

Si p-value < 0,05 → se rechaza H0
Si p-value > 0,05 → no se rechaza H0

Conclusión:
En base al resultado del test, se concluye que (completar según resultado):

Si p < 0,05: las variables son conjuntamente significativas y mejoran el modelo.
Si p > 0,05: las variables no aportan significativamente al modelo.

Pregunta 2.4- Volviendo a la regresión del salario en función de sales, roe y ros. Considere
dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos
clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de
significancia. Use el metodo que más le acomode.

In [ ]:
ceosal1$sales_D <- ceosal1$sales * ceosal1$sales_mayor_promedio
ceosal1$roe_D <- ceosal1$roe * ceosal1$sales_mayor_promedio
ceosal1$ros_D <- ceosal1$ros * ceosal1$sales_mayor_promedio

modelo_grupos <- lm(salary ~ sales + roe + ros + sales_mayor_promedio + sales_D + roe_D + ros_D, data = ceosal1)

summary(modelo_grupos)

anova(modelo1, modelo_grupos)


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_D + roe_D + ros_D, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1024.7  -456.7  -210.2    83.2 13712.4 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)  
(Intercept)           331.86179  352.76387   0.941   0.3480  
sales                   0.13362    0.06635   2.014   0.0454 *
roe                    25.63426   13.07068   1.961   0.0512 .
ros                    -0.47570    1.66943  -0.285   0.7760  
sales_mayor_promedio 1043.02106  578.05672   1.804   0.0727 .
sales_D                -0.13033    0.06731  -1.936   0.0542 .
roe_D                 -17.86857   28.02885  -0.638   0.5245  
ros_D                   1.55160    3.84565   0.403   0.6870  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1353 on 201 degrees of freedom
Multiple R-squared:  0.06028,	Adjusted R-squared:  0.02756 
F-statistic: 1.842 on 7 and 201

,Res.Df,RSS,Df,Sum of Sq,F,Pr(>F)
,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
1,205,379231123,NA,NA,NA,NA
2,201,368117757,4,11113366,1.517033,0.1985567


2.4. Se realiza un test F para evaluar si existen diferencias entre empresas con ventas mayores al promedio y aquellas con ventas menores o iguales al promedio.

Hipótesis:

H0: no hay diferencias entre los grupos
H1: existen diferencias

Regla de decisión:

Si p-value < 0,05 → se rechaza H0
Si p-value > 0,05 → no se rechaza H0

Conclusión:

En este caso, el p-value del test es 0.1986, el cual es mayor a 0,05.

Por lo tanto, no se rechaza la hipótesis nula, lo que indica que no existe evidencia estadísticamente significativa para afirmar que hay diferencias entre los grupos en la determinación del salario del CEO.

Pregunta 2.5- Usando la regresión del salario en función de sales, roe y ros, se distinguen los
grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de
empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada
grupo son iguales.

In [ ]:
ceosal1_indus <- subset(ceosal1, indus == 1)
ceosal1_finance <- subset(ceosal1, finance == 1)
ceosal1_consprod <- subset(ceosal1, consprod == 1)
ceosal1_utility <- subset(ceosal1, utility == 1)

modelo <- lm(salary ~ sales + roe + ros, data = ceosal1)

modelo_indus <- lm(salary ~ sales + roe + ros, data = ceosal1_indus)
modelo_finance <- lm(salary ~ sales + roe + ros, data = ceosal1_finance)
modelo_consprod <- lm(salary ~ sales + roe + ros, data = ceosal1_consprod)
modelo_utility <- lm(salary ~ sales + roe + ros, data = ceosal1_utility)

SSR_r <- sum(residuals(modelo)^2)

SSR_indus <- sum(residuals(modelo_indus)^2)
SSR_finance <- sum(residuals(modelo_finance)^2)
SSR_consprod <- sum(residuals(modelo_consprod)^2)
SSR_utility <- sum(residuals(modelo_utility)^2)

SSR_ur <- SSR_indus + SSR_finance + SSR_consprod + SSR_utility

G <- 4
k <- 3
n <- nrow(ceosal1)

gl1 <- (G - 1) * (k + 1)
gl2 <- n - G * (k + 1)

F_est <- ((SSR_r - SSR_ur) / gl1) / (SSR_ur / gl2)

F_critico <- qf(0.95, gl1, gl2)

F_est
F_critico

if (F_est > F_critico) {
  print("Se rechaza H0")
} else {
  print("No se rechaza H0")
}

[1] 1.398349

[1] 1.802614

[1] "No se rechaza H0"


2.5. Se aplica un test de Chow para evaluar si el modelo de regresión es igual entre los distintos sectores económicos.

H0: los modelos son iguales
H1: los modelos son distintos

Regla:

Si F_est > F_critico → se rechaza H0
Si F_est ≤ F_critico → no se rechaza H0

Conclusión:

Si F_est > F_critico → los modelos difieren entre sectores.
Si F_est ≤ F_critico → no hay evidencia de diferencias entre sectores.